test/unit-test/test_app_btc_common.c

```diff
@@ -316,6 +319,143 @@ static void _test_btc_common_encode_xpub(void** state)
         "yj6QKEuDXWvNbCDF98tgM");
 }
 
+typedef struct {
+    uint32_t threshold;
+    size_t xpubs_count;
+    const char* xpubs[15];
+    uint32_t keypath_change;
+    uint32_t keypath_address;
+    const char* expected_script_hex;
+} test_case_redeemscript_multisig_p2wsh;
+
+static void _test_btc_common_pkscript_from_multisig(void** state)
+{
+    // clang-format off
+    const test_case_redeemscript_multisig_p2wsh tests[] = {
```

Comment:
Add a test having an input with no xpubs (and `.xpubs_count == 0`) maybe

Reply:
added some negative tests.